### PR TITLE
Tagless common, model + repo

### PR DESCRIPTION
**Pull request by LordRaydenMK**
First commit moves existing stuff to `arrowio`. 

Second commit creates a new `tagless` for the tagless implementation. The common and model are basically copy -> paste. I also implemented the repo and repo interpreter.

### CODE REVIEW

File: src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class AccountRepositoryInMemory<F>(me: MonadError<F, Throwable>) : AccountRepository<F>,
+    MonadError<F, Throwable> by me {
+
+    private val repo by lazy { mutableMapOf<String, Account>() }
+
+    override fun query(no: String): Kind<F, Option<Account>> =
+        just(repo[no].toOption())
+
+    override fun store(a: Account): Kind<F, Account> {
+        repo[a.no] = a
```
Comment (pakoito) [inline src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt:22]: Side-effect. Is there any reason why MonadError and not MonadDefer?

**Review by pakoito — APPROVED** [review]
LGTM. One question.

File: src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt (hunk 1 above)
Comment (LordRaydenMK) [reply src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt:22]: It is what the Scala version uses. Maybe MonadError in cats is lazy?

File: src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    override fun all(): Kind<F, Sequence<Account>> =
+        just(repo.values.asSequence())
+
+    override fun balance(no: String): Kind<F, Balance> =
+        repo[no].toOption()
+            .map { it.balance }
+            .fold(
+                { throw IllegalArgumentException("Non existing account not $no") },
```
Comment (LordRaydenMK) [inline src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt]: Note to self: use raiseError

**Review by pakoito — APPROVED** [review]

**Pull request merged**